plugins/Actions/Archiver.php

```diff
@@ -468,35 +470,71 @@ protected function insertPageUrlsReports()
      */
     protected function archiveDayActionsGoals(int $rankingQueryLimit): void
     {
-        $this->archiveDayActionsGoalsPages($rankingQueryLimit,true);
-        $this->archiveDayActionsGoalsPages($rankingQueryLimit,false);
+        if (GeneralConfig::getConfigValue('disable_archive_actions_goals', $this->getProcessor()->getParams()->getSite()->getId())) {
+            return;
+        }
+
+        $goals = $this->getGoalsForSite($this->getProcessor()->getParams()->getSite()->getId());
+
+        if ($this->getProcessor()->getParams()->getSite()->isEcommerceEnabled()) {
+            $goals[] = 0;
+        }
```

Comment:
Shouldn't we also check if the goals plugin is actually enabled. In theory it could be disabled, causing the API call for `Goals.get` resulting in an error.

Also it might be easier to read when using a variable instead of repeating the chained method calls.

```suggestion
        $site = $this->getProcessor()->getParams()->getSite();

        if (!\Piwik\Common::isGoalPluginEnabled() || GeneralConfig::getConfigValue('disable_archive_actions_goals', $site->getId())) {
            return;
        }

        $goals = $this->getGoalsForSite($site->getId());

        if ($site->isEcommerceEnabled()) {
            $goals[] = 0;
        }
```